src/averageAges.js

```diff
@@ -55,7 +67,25 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const children = people.filter(person => (
+    people.find(mother => mother.name === person.mother) && (onlyWithSon
+      ? person.sex === 'm'
+      : true)
+  ));
+
+  const ageDiff = children.map(child => {
+    const mother = people.find(mom => mom.name === child.mother);
+
+    return child.born - mother.born;
+  });
+
+  return calculateAverageAge(ageDiff, children.length);
+}
+
+function calculateAverageAge(peopleAge, amountOfPeople) {
```

Comment:
Will be better to use `number` for count nouns
```suggestion
function calculateAverageAge(peopleAge, numberOfPeople) {
```

Reply:
Great job with moving repeated logic to separate function⭐️